### DISCUSSION TITLE
Support for Ubuntu LTS releases

### DISCUSSION THREAD

**Discussion by gaby** [FiloSottile/age#313] [Q&A]
I'm interesting in using `age` but noticed that it's only available in `Ubuntu 21.04`, are there any plans to add-it to Ubuntu latest LTS release (20.04 Focal)?

**Comment by FiloSottile** [comment]
You can use the binary downloads (https://dl.filippo.io/age/latest?for=linux/amd64) on any version of Ubuntu, but I don't think Ubuntu does backports of new packages to older versions. You can try opening a bug with them, though!

**Comment by gaby** [comment]
Yeah, you are right. Ubuntu won't allow new packages for a LTS already released. 

One way of getting packages into older Ubuntu releases is through [Launchpad.net](https://launchpad.net/). That's the way that the Ansible team does it. They are registered as a team in Launchpad which allows them to create a PPA (Personal Package Archive).

Example of Ansible PPA:

```
sudo apt update
sudo apt install software-properties-common
sudo add-apt-repository --yes --update ppa:ansible/ansible
sudo apt install ansible
```

**Reply by FiloSottile** [reply]
I would probably set up a PPA (also as a way to get the latest version on LTS releases) if someone contributed a PR for a GitHub Actions workflow that publishes the package 👍

**Reply by alerque** [reply]
I haven't seen that done. It would also be a little dodgy because PPAs expect packages to be GPG signed, and doing that in CI is always a compromise. That being said perhaps it doesn't matter much what key is used to sign and hence it can be a semi-disposable one just for the purpose. I'll keep my eyes on this issue though — if somebody does come up with (or points out) such a workflow I would *love* to copy it for another project where I tediously bump packages in a PPA by hand across 4 Ubuntu releases.

**Reply by gaby** [reply]
It seems no one has publish an action for doing this. From what I found it requires using `dput` as stated [here](https://help.launchpad.net/Packaging/PPA/Uploading). 

For example: 

```
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Run Install Commands
        run: |
          sudo apt-get install devscripts dput
```

**Comment by FiloSottile** [comment]
If someone were to compromise my GitHub account, we'd have bigger problems :)
